controllers/azurecluster_reconciler.go

```diff
@@ -110,6 +114,13 @@ func (r *azureClusterReconciler) Reconcile(ctx context.Context) error {
 		return errors.Wrapf(err, "failed to reconcile private dns")
 	}
 
+	// reconcile availability sets if there are no failure domains
+	if len(r.scope.FailureDomains()) == 0 {
```

Comment:
there is no need for this check, if there are no failure domains the reconcile should be a no-op as `AvailabilitySetSpecs()` will return an empty slice

Reply:
> if there are no failure domains the reconcile should be a no-op
IIUC, isn't it the opposite of this? As in, reconcile availability sets only when there are no failure domains for that location.

Reply:
sorry, if _there are_ failure domains the reconcile should be a no-op